app/src/main/java/a3/com/convo/fragments/GameFragment.java

```diff
@@ -53,64 +60,57 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
     public void onViewCreated(View view, Bundle savedInstanceState) {
         cardStack = (SwipeDeck) view.findViewById(R.id.cardStack);
 
+        // Overall game timer elements
+        final TextView tvTimer = (TextView) view.findViewById(R.id.tvTimer);
+        CountDownTimer timer = new CountDownTimer(Constants.GAME_TIME, Constants.TIMER_INTERVAL) {
+            @Override
+            public void onTick(long l) {
+                tvTimer.setText(
+                        String.format(context.getResources().getString(R.string.timer_format), TimeUnit.MILLISECONDS.toMinutes(l),
+                                TimeUnit.MILLISECONDS.toSeconds(l)
+                                        - TimeUnit.MINUTES.toSeconds(TimeUnit.MILLISECONDS.toMinutes(l)))
+                );
+            }
+
+            @Override
+            public void onFinish() {
+                tvTimer.setText(R.string.game_over);
+            }
+        };
+        timer.start();
+
         player1 = ParseUser.getCurrentUser();
         // pageLikes is guaranteed to be an array, but it's returned as an object anyway
-        p1Likes = (ArrayList<String>) player1.get("pageLikes");
+        p1Likes = (ArrayList<String>) player1.get(PAGE_LIKES);
 
         // get the second player and their likes
         ParseQuery<ParseUser> query = ParseUser.getQuery();
-        query.whereEqualTo("username", friend);
-        query.findInBackground(new FindCallback<ParseUser>() {
-            @Override
-            public void done(List<ParseUser> objects, ParseException e) {
-                if (e == null) {
-                    player2 = objects.get(0);
-                    p2Likes = (ArrayList<String>) player2.get("pageLikes");
-
-                    // put together both player's likes and shuffle them
-                    allLikes = new ArrayList<>();
-                    allLikes.addAll(p1Likes);
-                    allLikes.addAll(p2Likes);
-                    Collections.shuffle(allLikes);
-
-                    adapter = new CardAdapter(allLikes, context);
-                    cardStack.setAdapter(adapter);
-                } else {
-                    e.printStackTrace();
+
+        if (!friend.equals("")) {
+            query.getInBackground(friend, new GetCallback<ParseUser>() {
+                @Override
+                public void done(ParseUser object, ParseException e) {
+                    if (e == null) {
```

Comment:
Can we check if object is not null and not empty, instead of checking if exception is null?